### PR TITLE
Add tps metrics and tidb-server status rest api

### PR DESCRIPTION
**Pull request by shenli**

### CODE REVIEW

File: metric/metric.go [hunk 1]
```diff
@@ -54,3 +54,56 @@ func RunMetric(interval time.Duration) {
...
+	// Get current tps
+	Get() int64
+}
+
+// Simple tps metrics
+// Accumulate txn count in a second and reset the counter at each second.
+type tpsMetrics struct {
+	meter metrics.Counter
+	tps   int64
+	mutex sync.Mutex
```
Comment (ngaut) [inline metric/metric.go]: s/mutex/mu

File: metric/metric.go [hunk 2]
```diff
@@ -54,3 +54,56 @@ func RunMetric(interval time.Duration) {
...
+		tm.tick()
+		time.Sleep(1 * time.Second)
+	}
+}
+
+// NewTPSMetrics creates a tpsMetrics and starts its ticker.
+func NewTPSMetrics() TPSMetrics {
+	m := &tpsMetrics{
+		meter: metrics.NewCounter(),
+		tps:   0,
```
Comment (ngaut) [inline metric/metric.go]: Default value of tps is 0.

File: metric/metric.go [hunk 3]
```diff
@@ -54,3 +54,56 @@ func RunMetric(interval time.Duration) {
...
+func (tm *tpsMetrics) Get() int64 {
+	tm.mutex.Lock()
+	defer tm.mutex.Unlock()
+	return tm.tps
+}
+
+func (tm *tpsMetrics) tick() {
+	tm.mutex.Lock()
+	defer tm.mutex.Unlock()
+	t := tm.meter.Count()
```
Comment (ngaut) [inline metric/metric.go]: We should use a monotonic increment counter. No need to clear.

Comment (shenli) [inline metric/metric.go]: Than we will have to maintain previous value of the counter.

**Comment by shenli** [comment]
@ngaut PTAL

**Comment by ngaut** [comment]
@iamxy PTAL

**Comment by iamxy** [comment]
LGTM

**Comment by shenli** [comment]
@ngaut @coocood PTAL

**Comment by coocood** [comment]
LGTM

**Pull request merged**